From 634017950c1c920d0de63fffa5c52e621de1d603 Mon Sep 17 00:00:00 2001
From: Wang Mingyu <wangmy@fujitsu.com>
Date: Wed, 26 Jun 2024 07:41:01 +0000
Subject: Fix incompatible pointer type error with gcc option
 -Wincompatible-pointer-types

lib32-openldap do_compile failure with gcc-14:
| tls_g.c:971:57: error: passing argument 4 of 'gnutls_fingerprint' from incompatible pointer type [-Wincompatible-pointer-types]
|   971 |                                         keyhash.bv_val, &keyhash.bv_len ) < 0 ) {
|       |                                                         ^~~~~~~~~~~~~~~
|       |                                                         |
|       |                                                         ber_len_t * {aka long unsigned int *}
| In file included from tls_g.c:44:
| /usr/include/gnutls/gnutls.h:2406:32: note: expected 'size_t *' {aka 'unsigned int *'} but argument is of type 'ber_len_t *' {aka 'long unsigned int *'}
|  2406 |                        size_t *result_size);
|       |                        ^~~

Upstream-Status: Submitted

Signed-off-by: Wang Mingyu <wangmy@fujitsu.com>
---
 libraries/libldap/tls_g.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libraries/libldap/tls_g.c b/libraries/libldap/tls_g.c
index 7c23875..b4b487d 100644
--- a/libraries/libldap/tls_g.c
+++ b/libraries/libldap/tls_g.c
@@ -968,7 +968,7 @@ tlsg_session_pinning( LDAP *ld, tls_session *sess, char *hashalg, struct berval
 		keyhash.bv_len = gnutls_hash_get_len( alg );
 		keyhash.bv_val = LDAP_MALLOC( keyhash.bv_len );
 		if ( !keyhash.bv_val || gnutls_fingerprint( alg, &key,
-					keyhash.bv_val, &keyhash.bv_len ) < 0 ) {
+					keyhash.bv_val, (size_t *)&keyhash.bv_len ) < 0 ) {
 			goto done;
 		}
 	} else {
-- 
2.34.1

